### PR TITLE
Fix 1.10 config menu

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Reverts the config style back to Pre RS merge to allow for forge's config GUI to be used.
- Yes there are 3 lint suppressions, this was just copied from old branch with a touch-up, feel free to add to it if you'd like.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 1]
```diff
@@ -134,14 +134,14 @@ public AIState gather()
...
-            maximalGatherCount = Configurations.requestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.requestSystem.maximalBuildingsToGather - Configurations.requestSystem.minimalBuildingsToGather));
+            maximalGatherCount = Configurations.RequestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.RequestSystem.maximalBuildingsToGather - Configurations.RequestSystem.minimalBuildingsToGather));
         }
 
         if (gatherTarget == null)
         {
             if (gatherCount == maximalGatherCount)
             {
-                maximalGatherCount = Configurations.requestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.requestSystem.maximalBuildingsToGather - Configurations.requestSystem.minimalBuildingsToGather));
+                maximalGatherCount = Configurations.RequestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.RequestSystem.maximalBuildingsToGather - Configurations.RequestSystem.minimalBuildingsToGather));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: MAJOR Split this 244 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 2]
```diff
@@ -134,14 +134,14 @@ public AIState gather()
     {
         if (maximalGatherCount < 0)
         {
-            maximalGatherCount = Configurations.requestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.requestSystem.maximalBuildingsToGather - Configurations.requestSystem.minimalBuildingsToGather));
+            maximalGatherCount = Configurations.RequestSystem.minimalBuildingsToGather + worker.getRandom().nextInt(Math.max(1, Configurations.RequestSystem.maximalBuildingsToGather - Configurations.RequestSystem.minimalBuildingsToGather));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: MAJOR Split this 240 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/Pathfinding.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
     private static final ThreadPoolExecutor executor;
     static
     {
-        executor = new ThreadPoolExecutor(1, Configurations.pathfinding.pathfindingMaxThreadCount, 10, TimeUnit.SECONDS, jobQueue);
+        executor = new ThreadPoolExecutor(1, Configurations.Pathfinding.pathfindingMaxThreadCount, 10, TimeUnit.SECONDS, jobQueue);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/Pathfinding.java]: MAJOR Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        return Collections.emptySet();
+    }
+    
+    /**
+     * @deprecated This was never fully implemented and will be removed in the future.
+     */
+    @Override
+    @Deprecated
+    public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement element)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 203 issues
* ![CRITICAL][CRITICAL] 12 critical
* ![MAJOR][MAJOR] 158 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L916](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L916): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L921](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L921): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L188](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L188): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [StandardRetryingRequestResolver.java#L43](https://github.com/Minecolonies/minecolonies/blob/2c48643343dfd792ffe29147fb4878ffedad3a45/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardRetryingRequestResolver.java#L43): Remove this call from a constructor to the overridable "updateManager" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 203 issues
* ![CRITICAL][CRITICAL] 12 critical
* ![MAJOR][MAJOR] 158 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L916](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L916): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L921](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L921): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L188](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L188): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [StandardRetryingRequestResolver.java#L43](https://github.com/Minecolonies/minecolonies/blob/8b5fce58d26fefddd5d8ee954a0d337f979dd5b9/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardRetryingRequestResolver.java#L43): Remove this call from a constructor to the overridable "updateManager" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/api/java/com/minecolonies/api/configuration/ConfigurationHandler.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    private static Configuration config;
+
+    public static final String CATEGORY_GAMEPLAY    = "gameplay";
+    public static final String CATEGORY_PATHFINDING = "pathfinding";
+    public static final String CATEGORY_NAMES       = "names";
+    public static final String CATEGORY_REQUEST     = "requestsystem";
+
+    private static final String FORMAT_RANGE = "%s (range: %s ~ %s, default: %s)";
+
+    public ConfigurationHandler()
```
[the remark this answers is not in the document]
Comment (Asherslab) [reply src/api/java/com/minecolonies/api/configuration/ConfigurationHandler.java]: it has a comment in it saying why it's there, also it was there before /shrug

**Comment by Raycoms** [comment]
:shipit: looks good for me.

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1958/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        return Collections.emptySet();
+    }
+    
+    /**
+     * @deprecated This was never fully implemented and will be removed in the future.
+     */
+    @Override
+    @Deprecated
+    public RuntimeOptionGuiHandler getHandlerFor(final RuntimeOptionCategoryElement element)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 203 issues
* ![CRITICAL][CRITICAL] 12 critical
* ![MAJOR][MAJOR] 158 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L916](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L916): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L921](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L921): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L188](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L188): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [StandardRetryingRequestResolver.java#L43](https://github.com/Minecolonies/minecolonies/blob/3edaad91f4233f2416cbb690d69625d538f166cd/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardRetryingRequestResolver.java#L43): Remove this call from a constructor to the overridable "updateManager" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**